apiserver/facades/client/bundle/bundle.go

```diff
@@ -143,16 +181,47 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 	if err != nil {
 		return results, err
 	}
-	results.Changes = make([]*params.BundleChange, len(changes))
-	for i, c := range changes {
-		results.Changes[i] = &params.BundleChange{
-			Id:       c.Id(),
-			Method:   c.Method(),
-			Args:     c.GUIArgs(),
-			Requires: c.Requires(),
-		}
+	err = postProcess(changes, &results)
+	return results, err
+}
+
+// GetChanges returns the list of changes required to deploy the given bundle
+// data. The changes are sorted by requirements, so that they can be applied in
+// order.
+func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleChangesResults, error) {
+	vs := validators{
+		verifyConstraints: func(s string) error {
+			_, err := constraints.Parse(s)
+			return err
+		},
+		verifyStorage: func(s string) error {
+			_, err := storage.ParseConstraints(s)
+			return err
+		},
+		verifyDevices: func(s string) error {
+			_, err := devices.ParseConstraints(s)
+			return err
+		},
 	}
-	return results, nil
+	return getChanges(args, vs, func(changes []bundlechanges.Change, results *params.BundleChangesResults) error {
+		results.Changes = make([]*params.BundleChange, len(changes))
+		for i, c := range changes {
+			var guiArgs []interface{}
+			switch c := c.(type) {
```

Comment:
I wonder if it is needed now... I mean we are bringing in the version of bundlechanges that will have the new GUIArgsWith... so we can always call it...

Could you please explain the rationale for the 2 code paths here?

Reply:
Because here in the `[] change`, the change is not always needs to run `GUIArgsWithDevices` but only for `ApplicationChange`.